lib/SIL/SILVerifier.cpp

```diff
@@ -410,6 +410,130 @@ void verifyKeyPathComponent(SILModule &M,
   baseTy = componentTy;
 }
 
+/// Check if according to the SIL language model this memory /must only/ be used
+/// immutably. Today this is only applied to in_guaranteed arguments and
+/// open_existential_addr. We should expand it as needed.
+struct ImmutableAddressUseVerifier {
+  SmallVector<Operand *, 32> worklist;
```

Comment:
NOTE: I am going to use this in a forthcoming commit to look through certain instructions recursively.